src/matter/interaction/SubscriptionHandler.ts

```diff
@@ -28,30 +29,31 @@ export class SubscriptionHandler {
         private readonly minIntervalFloorMs: number,
         private readonly maxIntervalCeilingMs: number,
     ) {
-        attributes.forEach(({ attribute }) => attribute.addListener(this.listener));
-        this.intervalId = setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs);
+        attributes.forEach(({ attribute }) => attribute.addMatterListener(this.listener));
+        this.updateTimer = Time.get().getTimer(this.minIntervalFloorMs, () => this.sendUpdate()).start();
+        this.updateTimer.start();
     }
 
     async sendUpdate() {
         const now = Date.now();
         const timeSinceLastUpdateMs = now - this.lastUpdateTimeMs;
         if (timeSinceLastUpdateMs < this.minIntervalFloorMs) {
-            clearTimeout(this.intervalId);
-            this.intervalId = setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs - timeSinceLastUpdateMs);
+            this.updateTimer.stop();
+            this.updateTimer = Time.get().getTimer(setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs - timeSinceLastUpdateMs);
             return;
         }
 
         const values = this.attributes.map(({ attribute, path }) => ({ path, valueVersion: attribute.getWithVersion(), template: attribute.template }));
         await this.sendUpdateMessage(values);
         this.lastUpdateTimeMs = now;
 
-        clearTimeout(this.intervalId);
-        this.intervalId = setTimeout(() => this.sendUpdate(), this.maxIntervalCeilingMs);
+        clearTimeout(this.updateTimer);
+        this.updateTimer = setTimeout(() => this.sendUpdate(), this.maxIntervalCeilingMs);
```

Comment:
also here use timer.stop() to stop and not settimeout? (or is that not based on new Time stuff?)